src/utils/Math.vy

```diff
@@ -0,0 +1,96 @@
+# @version ^0.3.7
+
+
+@external
+@view
+def wad_cbrt(x: uint256) -> uint256:
+    """
+    @notice Calculate the cubic root of a number in 1e18 precision
+    @dev Consumes around 1500 gas units
+    @param x The number to calculate the cubic root of
+    @return The cubic root of the number
+    """
+
+    # Since this cubic root is for numbers at 1e18 base, we need to scale the
+    # input by 1e36 to increase precision. This will overflow for very large
+    # numbers. So conditionally sacrifice precision.
+
+    xx: uint256 = 0
+    if x >= 115792089237316195423570985008687907853269 * 10**18:
```

Comment:
Also, `10**18` could be defined as another constant to save gas, since it's reachable multiple times if I'm correct in the contract logic.

Reply:
Forget about that comment since the number is precomputed anyways.

Reply:
No it doesn't cost additional gas actually (compiler optimises it).

Reply:
yeah I realised that as well - that's why I wrote to forget about this comment lol